### PR TITLE
Proof read of the Soto 5 docs

### PR DESCRIPTION
**Pull request by 0xTim**
Plenty of things for discussion.

Do we Americanize the docs? That's a habit I've picked up from writing the Vapor book.

Using the same `AWSClient` across the application and all request threads might cause issues since it's a class, so it will be the same instance across all. The services should get a copy since they're `struct`s but the 'Vapor way' for other stuff is to create a new instance for each request and the application

### CODE REVIEW

**Comment by adam-fowler** [comment]
> Do we Americanize the docs? That's a habit I've picked up from writing the Vapor book.
> 
I'm ok with Americanization

> Using the same `AWSClient` across the application and all request threads might cause issues since it's a class, so it will be the same instance across all. The services should get a copy since they're `struct`s but the 'Vapor way' for other stuff is to create a new instance for each request and the application

`AWSClient` holds `AsyncHTTPClient.HTTPClient` and a `CredentialProvider` which both need a shutdown. Passing these around in a struct means there isn't any object that own these objects. I would see `AWSClient` being similar to `Vapor.Application` and `AsyncHTTPClient.HTTPClient` which are both classes.

File: documentation/DynamoDB and Codable.md [hunk 1]
```diff
@@ -40,12 +48,16 @@ let input = DynamoDB.PutItemInput(
...
+
 can be reduced to 
+
 ```swift
 let input = DynamoDB.PutItemCodableInput(item: person, tableName: "my-table")
 let output = try dynamoDB.putItem(input).wait()
 ```
-Similarly when using `DynamoDB.GetItem` the code below which parses an `AttributeValue` dictionary returned from `GetItem`
+
+Similarly when using `DynamoDB.GetItemInput(key:tableName:)`, the code below which parses an `AttributeValue` dictionary returned
```
Comment (adam-fowler) [inline documentation/DynamoDB and Codable.md:59]: DynamoDB.getItem is the function. DynamoDB.GetItemInput is the input type to that function. I would suggest we use the function name.

File: documentation/S3 Multipart Upload.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+S3 has a series of multipart upload operations. These can be used to upload an object to S3 in multiple parts. If your object is larger than 5GB you are required to use the multipart operations for uploading, but multipart also has the advantage that if one part fails to upload you don't need to re-upload the whole object, just the parts that failed. Finally you can upload multiple parts at the same time and thus improve your upload speed. The multiple part uploading isn't support by Soto at the moment, but it is on the list of improvements to make. You can read more about multipart upload [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).
```
Comment (adam-fowler) [inline documentation/S3 Multipart Upload.md:3]: Maybe we should add "You can read Amazon's documentation on multipart upload here"

**Review by adam-fowler — COMMENTED** [review]
These all look good, just a couple of minor changes

File: documentation/S3 Multipart Upload.md (hunk 2 above)
Comment (0xTim) [reply documentation/S3 Multipart Upload.md:3]: Do you mean something like

```suggestion
S3 has a series of multipart upload operations. These can be used to upload an object to S3 in multiple parts. If your object is larger than 5GB you are required to use the multipart operations for uploading, but multipart also has the advantage that if one part fails to upload you don't need to re-upload the whole object, just the parts that failed. Finally you can upload multiple parts at the same time and thus improve your upload speed. The multiple part uploading isn't support by Soto at the moment, but it is on the list of improvements to make. You can read  Amazon's documentation on multipart upload [here](https://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html).
```

Comment (adam-fowler) [reply documentation/S3 Multipart Upload.md:3]: yep

**Comment by 0xTim** [comment]
Requested changes made

**Review by adam-fowler — APPROVED** [review]
I think we're good here
thanks

**Pull request merged**